modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFileBootstrap.pm

```diff
@@ -0,0 +1,226 @@
+      my $end = int($bin_size * ($i+1));
+      $end = $length if $end > $length;
+      push(@bed, join("\t", ($name, $start, $end))."\n");
+    }
+  }
+  my $path = $genome->genome_summary_bed_path();
+  $path->spew(@bed);
+
+}
+
+sub prepare_sql {
+
+  my ($self, $genome_id) = @_;
+      # Query suggested by Andy that reproduces assembly reports that we used to get from NCBI
+      #   # (example of NCBI assembly report: ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA_000001405.27_GRCh38.p12/GCA_000001405.27_GRCh38.p12_assembly_report.txt)
+  
+    return "
+    select sr.name, CASE WHEN IFNULL(sra3.value, 0) = 0 AND cs.name = 'chromosome' THEN 'assembled-molecule' ELSE 'unlocalized-scaffold' END, sr.name, cs.name, srs1.synonym, '=', srs2.synonym, 'Primary Assembly', sr.length, srs3.synonym
+    from coord_system cs
+    join seq_region sr on (cs.coord_system_id = sr.coord_system_id)
+    join seq_region_attrib sra1 on (sr.seq_region_id = sra1.seq_region_id)
+    join attrib_type at1 on (sra1.attrib_type_id = at1.attrib_type_id)
+    left join seq_region_attrib sra2 on (sr.seq_region_id = sra2.seq_region_id and sra2.attrib_type_id = (select attrib_type_id from attrib_type where code = 'karyotype_rank'))
+    left join seq_region_synonym srs1 on (sr.seq_region_id = srs1.seq_region_id and srs1.external_db_id = (select external_db_id from external_db where db_name = 'INSDC'))
+    left join seq_region_synonym srs2 on (sr.seq_region_id = srs2.seq_region_id and srs2.external_db_id = (select external_db_id from external_db where db_name = 'RefSeq_genomic'))
+    left join seq_region_synonym srs3 on (sr.seq_region_id = srs3.seq_region_id and srs3.external_db_id = (select external_db_id from external_db where db_name = 'UCSC'))
+    left join seq_region_attrib sra3 on (sr.seq_region_id = sra3.seq_region_id and sra3.attrib_type_id = (select attrib_type_id from attrib_type where code = 'non_ref'))
+    where at1.code = 'toplevel'
+    and cs.species_id = $genome_id
+    order by IFNULL(CONVERT(sra2.value, UNSIGNED), 5e6) asc, sr.name;
+  ";
+
+
+}
+
+
+sub write_column_headers {
+  my ($self, $file_handler) = @_;
+  my $column_names = "# Sequence-Name   Sequence-Role   Assigned-Molecule       Assigned-Molecule-Location/Type GenBank-Accn    Relationship    RefSeq-Accn     Assembly-Unit   Sequence-Length UCSC-style-name\n";
+  print $file_handler $column_names;
+}
+
+sub write_report {
+  my ($self, $file_handler, $data) = @_;
+
+  foreach my $row (@$data) {
+    my @row_of_strings = map { defined $_ ? $_ : '-' } @{$row};
+    print $file_handler join("\t", @row_of_strings) . "\n";
+  }
+}
+
+
+
+
+sub write_output {
+  my ($self) = @_;
+  my $species        = $self->param('species');
+  my $group          = $self->param('group');
+  my $current_step   = $self->param('current_step');
+  
+  if ($current_step eq 'bootstrap') {
+        my @steps     = @{$self->param('step')};
+        my %each_flow = (
+             'transcripts' => 1,
```

Comment:
It's best to leave flow `1` alone, and start at (at least) `2`. This is because pipelines automatically flow down pipe 1; most of the time that's not an issue, and I don't think it'd cause any problems currently. But it can cause unexpected behaviour, particularly if you have conditional dataflow, so it's best to get in the habit of avoiding it.

Reply:
sure will follow